3.3 blog post improvements (#109)
sourcegraph/about: blogposts/announcing-sourcegraph-3.3.md

published: true
---

We're excited to announce Sourcegraph 3.3. Sourcegraph is a code search and navigation tool (self-hosted, open-source, and cross-repository). Read on for the details, and thanks to our customers and community for reporting issues and providing feedback. You’re helping to make each Sourcegraph release the best one yet!

<div style="padding-left: 2rem">

[**🛠️ Config-based repository selection**](#config-based-repository-selection)<br />
Define add/exclude lists in external services instead of enabling/disabling.

**[🧯 Jump to Sentry issue from your code](#new-sentry-integration-with-the-sentry-sourcegraph-extension)**<br/>

> Error and exception handling code, annotated with Sentry links.

Reviewer: ```suggestion
Your error and exception handling code, annotated with Sentry links.
```

**[🔌 Bitbucket Server integration](#bitbucket-server-integration)**<br />

> Better PR code reviews on Bitbucket Server with hover tool-tips and "go to definition".

Reviewer: ```suggestion
Better PR code reviews on Bitbucket Server with hover tooltips and "go to definition".
```

We don't usually hyphenate tooltips

**[✔️Grant permissions on your code host](#grant-permissions-on-your-code-host)**<br />
When using code hosts other than GitHub.com.

[**🧭 "Go to definition" using dependency graph**](#go-to-definition-using-dependency-graph)<br />

> `import` / `require` statements used to help find the right definition.

Reviewer: ```suggestion
`import` / `require` statements are used to help find the right definition.
```

(I know I suggested the original wording without "are".)

[**🦅 Swift code navigation**](#swift-code-navigation)<br />
Fly around Swift code on Sourcegraph.com or your code host.

[**🗄 Symbols for the current file**](#symbols-for-the-current-file)<br />
The symbols sidebar is now scoped to the current file/directory you’re viewing.

<!--
TODO(ryan): Show once 3.3 is available on DigitalOcean
[**🚢 Deployments secure by default with TLS pre-configured**](#deployments)<br />
AWS and DigitalOcean deployments with TLS via self-signed certs.
-->

[**🔎 Handle common search queries containing (unintended) invalid regexp**](#handle-common-search-queries-containing-unintended-invalid-regexp)<br />

> Searching for function call sites e.g. `throw(` now work as expected.

Reviewer: ```suggestion
Searching for function call sites such as `open(` now work as expected.
```

I recommend `such as` instead of `e.g.` because `e.g.` usually is offset by commas, which would interfere with the example. And `open(` not `throw(` because the latter is usually not followed by parens, like `throw new Error('foo')` not `throw('foo')`.

> [**📣 Making Sourcegraph a part of your dev tools stack**](#making-sourcegraph-a-part-of-your-dev-tools-stack)<br />

Reviewer: ```suggestion
[**📣 Tips for making Sourcegraph a part of your dev tools stack**](#making-sourcegraph-a-part-of-your-dev-tools-stack)<br />
```

> Tips for integrating Sourcegraph and code search into your team's workflow.

Reviewer: ```suggestion
Documentation on integrating Sourcegraph and code search into your team's workflow.
```

[**📝 Changelog**](#330-changelog)<br />
Every detail that changed in this release

[**🎖️ Thank you**](#thank-you)<br />

> Sourcegraph couldn’t be what it is without the community

Reviewer: Be consistent about whether the sentences after the links have a period or not

Additional languages will be added as part of each new release and if you want a particular language added, create an issue for the [sourcegraph-sentry repository](https://github.com/sourcegraph/sentry/issues).

## Bitbucket Server integration

<p class="container">
  <div style="padding:56.25% 0 0 0;position:relative;">
    <iframe src="https://player.vimeo.com/video/331821650?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
  </div>
  <p style="text-align: center"><a href="https://vimeo.com/331821650" target="_blank">View on Vimeo</a></p>
</p>

> Get IDE-like features on your next pull request review inline on Bitbucket Server by installing the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension), then enabling it for your Bitbucket Server URL.

Reviewer: ```suggestion
Get IDE-like features on your next pull request review inline on Bitbucket Server by installing the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension), then granting permissions for your Bitbucket Server domain.
```

## Grant permissions on your code host

Granting permissions in the Sourcegraph browser extension is necessary when using anything except GitHub.com. It's now easier to discover:

<div style="text-align: center">

  ![](/blog/sourcegraph-browser-extension-enable-prompt.png)
  
</div>

## "Go to definition" using dependency graph

Definitions are now filtered using `import` / `require` statements to reduce false matches and improve accuracy.

![](/blog/filtered-imports.png)

## Swift code navigation

<p class="container">
  <div style="padding:56.25% 0 0 0;position:relative;">
    <iframe src="https://player.vimeo.com/video/331859021?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
  </div>
  <p style="text-align: center"><a href="https://vimeo.com/331859021" target="_blank">View on Vimeo</a></p>
</p>

> See it in action on [Sourcegraph.com](https://sourcegraph.com/github.com/googleprojectzero/fuzzilli/-/blob/Sources/Fuzzilli/Configuration.swift#L64:27) or enable it for all users on your private instance by adding it to the `extensions` object in **Site admin > global settings** (path is `/site-admin/global-settings`):

Reviewer: ```suggestion
See it in action on [Sourcegraph.com](https://sourcegraph.com/github.com/googleprojectzero/fuzzilli/-/blob/Sources/Fuzzilli/Configuration.swift#L64:27) or use it on your own instance for yourself or [for all users](https://docs.sourcegraph.com/admin/extensions#enable-an-extension-for-all-users):
```

```json
..."extensions": {
    ...

>     "sourcegraph/swift": true

Reviewer: Delete this JSON example now that I added the link to https://docs.sourcegraph.com/admin/extensions#enable-an-extension-for-all-users above

Terraform plans for Google Cloud and Azure are being scheduled for upcoming releases. Let us know via [tweet](https://twittercom/srcgraph) or [GitHub issue](https://github.com/sourcegraph/sourcegraph/issues/new?assignees=&labels=&template=feature_request.md&title=) which cloud provider you want support for next, along with any special instructions for that provider.
-->

## Handle common search queries containing (unintended) invalid regexp

> Using [regexp in search queries](https://docs.sourcegraph.com/user/search/queries) is powerful, but problematic prior to 3.3 in cases such as searching for function call sites, e.g. `throw(`. This pattern is now treated as a literal string (as expected).

Reviewer: ```suggestion
Using [regexps in search queries](https://docs.sourcegraph.com/user/search/queries) is powerful, but sometimes you forget and just want to find matches for `open(`. Previously, this was treated as a regexp error, but now Sourcegraph does The Right Thing™ and looks for the literal match when there is no ambiguity.
```
